set.go

```diff
@@ -324,13 +324,19 @@ func (s *Set) NewSummary(name string) *Summary {
 //
 // The returned summary is safe to use from concurrent goroutines.
 func (s *Set) NewSummaryExt(name string, window time.Duration, quantiles []float64) *Summary {
+	if err := validateMetric(name); err != nil {
+		panic(fmt.Errorf("BUG: invalid metric name %q: %s", name, err))
+	}
 	sm := newSummary(window, quantiles)
-	s.registerMetric(name, sm)
+
+	s.mu.Lock()
+	// defer will unlock in case of panic
+	defer s.mu.Unlock()
+
+	s.mustRegister(name, sm)
 	registerSummary(sm)
```

Comment:
Could you rename this function to `registerSummaryLocked` and verify that `s.mu` lock is locked at all the call sites for this function?